### ISSUE TITLE
Forced to attack creatures can't attack opponent with Propaganda even if attacker can pay and wishes to 

### ISSUE THREAD

**Issue by Alex-Vasile** [magefree/mage#9600]
Opponent has [[Propaganda]] out. I have a creature that's force to attack during a combat. I am unable to choose the controller of the Propaganda as the player I'm attacking even if I can pay the mana cost.

from discord.

**Comment by jeffwadsworth** [comment]
What effect/card is forcing the creature to attack and in what timestamp order? I tested with [[Curse of the Nightly Hunt]] and it worked.  The timestamp order was the curse and then Propaganda.  The targeted player of the curse was the AI.  I also tested from the AI side (it targeted me with the curse and it had Propaganda on the BF) and my creature was able to attack it after payment.  But, I did not test a multiplayer set up with 3 or more players.  Perhaps it happens in that scenario.

**Comment by jeffwadsworth** [comment]
[[Curse of the Nightly Hunt]]

**Comment by Grath** [comment]
From my (limited) interactions with forced attacks in XMage, I would expect this to only be reproducible in a multiplayer scenario where one player is forced to attack, at least one opponent has a Propaganda, and at least one opponent doesn't.

**Comment by jeffwadsworth** [comment]
Yeah, testing this would be easier if someone has a friend they can try it on.  Then post a picture of the board state and log, etc.

**Comment by Alex-Vasile** [comment]
I added a month of two ago a 4 player debug game with limited range of influence that can be accessed if you launch the client with the debug flag.

**Comment by Grath** [comment]
![image](https://user-images.githubusercontent.com/1895280/193352739-477e0498-9615-40a9-8bf5-24cf30dd47f5.png)
Retested with four players/one propaganda, two non, you're presented with a choice of which non-Propaganda player to attack but cannot click on the player with Propaganda.

**Comment by jeffwadsworth** [comment]
> I added a month of two ago a 4 player debug game with limited range of influence that can be accessed if you launch the client with the debug flag.

It would be great to have that entered into the wiki.  https://github.com/magefree/mage/wiki/Development-Testing-Tools

**Comment by Alex-Vasile** [comment]
> > I added a month of two ago a 4 player debug game with limited range of influence that can be accessed if you launch the client with the debug flag.
> 
> 
> 
> It would be great to have that entered into the wiki.  https://github.com/magefree/mage/wiki/Development-Testing-Tools

Good point. I forgot about that. I will update the wiki today.

**Comment by Alex-Vasile** [comment]
> ![image](https://user-images.githubusercontent.com/1895280/193351108-57cfa243-838e-4068-b2a3-26a4c0618afd.png)
> 
> Reproducible precisely as I suggested, when I clicked to combat it automatically chose to attack Grath2 because Grath has Propaganda. I was not given the opportunity to undo the attack and chose to attack Grath through propaganda, but that may be possible if there's a mix of creatures who are forced to attack and creatures who have optional attacks?
> 
> 
> 
> ![image](https://user-images.githubusercontent.com/1895280/193351591-87806858-2940-43ff-9d1b-a81e72d5af32.png)
> 
> When both opponents have effects that prevent a forced attack, it correctly does not force an attack and allows you to choose to attack an opponent if you wish to pay for the attack.
> 
> 
> 
> 

What is your auto-choose level set to?

**Comment by Grath** [comment]
Auto-choose is set to Most; but with my later comment on the 4-player scenario it didn't auto-choose and doesn't allow me to choose to attempt to attack the player with Propaganda. You can see the untapped Black Lotus so I absolutely had the mana for it.

**Comment by Alex-Vasile** [comment]
> Auto-choose is set to Most; but with my later comment on the 4-player scenario it didn't auto-choose and doesn't allow me to choose to attempt to attack the player with Propaganda. You can see the untapped Black Lotus so I absolutely had the mana for it.

Alright. What happens if you set it to off?

**Comment by Grath** [comment]
> > Auto-choose is set to Most; but with my later comment on the 4-player scenario it didn't auto-choose and doesn't allow me to choose to attempt to attack the player with Propaganda. You can see the untapped Black Lotus so I absolutely had the mana for it.
> 
> Alright. What happens if you set it to off?

![image](https://user-images.githubusercontent.com/1895280/193362769-39e767ba-4fa1-4296-89ac-d76ea41946da.png)
Immediately attacked the non-Propaganda opponent, no option whether or not to attack/where to attack.
